primitives/python/qrmi_primitives/pulser_qrmi_backend/pulser_qrmi_backend/backend.py

```diff
@@ -0,0 +1,131 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright 2025 Pasqal, IBM. All Rights Reserved.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from __future__ import annotations
+
+import json
+import logging
+import typing
+
+import pulser
+from pulser.backend.remote import (
+    BatchStatus,
+    JobParams,
+    JobStatus,
+    RemoteConnection,
+    RemoteResults,
+)
+from pulser.backend.results import Results
+from pulser.devices import Device
+
+from qrmi import Payload, QuantumResource  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class PulserQRMIConnection(RemoteConnection):
+    """A connection to Pasqal Cloud, to submit Sequences to QPUs."""
+
+    def __init__(self, qrmi: QuantumResource) -> None:
+        self._qrmi = qrmi
+
+    def supports_open_batch(self) -> bool:
+        """Flag to confirm this class doesn't support open batch creation."""
+        return False
+
+    def fetch_available_devices(self) -> dict[str, pulser.devices.Device]:
+        target = self._qrmi.target()
+        target = json.loads(target.value)
+        dev = Device(**target)
+        return {dev.name: dev}
+
+    def _fetch_result(
+        self, batch_id: str, job_ids: list[str] | None
+    ) -> pulser.Sequence[Results]:
+        raise NotImplementedError("Not applicable to current design")
+
+    def _get_batch_status(self, batch_id: str) -> BatchStatus:
+        raise NotImplementedError("Not applicable to current design")
+
+    def _query_job_progress(
+        self, batch_id: str
+    ) -> typing.Mapping[str, tuple[JobStatus, Results | None]]:
+        raise NotImplementedError("Not applicable to current design")
```

Comment:
How do you get away without any of these? Are you planning on implementing them later? `RemoteResults` relies on these methods to retrieve the results

Reply:
yeah, this is not really compatible with the design of qrmi...
Due to the device deserialization issues I could not run yet. working on it.
If truly required, then I cannot inherit from pulsers Backend.

for us doing `run(..., wait=True)` is enforced

Reply:
`RemoteBackend` specifies the return type of `run()` to be `RemoteResults`. What you can do is make your own subclass of `RemoteResults` to deal with the methods that requires these calls and you should be good